eda/src/autogluon/eda/auto/simple.py

```diff
@@ -924,3 +940,75 @@ def missing_values_analysis(
             MissingValues(graph_type=graph_type, **chart_args),
         ],
     )
+
+
+def explain_rows(
+    train_data: pd.DataFrame,
+    model: TabularPredictor,
+    rows: pd.DataFrame,
+    display_rows: bool = False,
+    plot: Optional[str] = "force",
+    baseline_sample: int = 100,
+    return_state: bool = False,
+    **kwargs,
+) -> Optional[AnalysisState]:
+    """
+    Kernel SHAP is a method that uses a special weighted linear regression
+    to compute the importance of each feature. The computed importance values
+    are Shapley values from game theory and also coefficients from a local linear
+    regression values analysis for the given rows.
+
+    The results are rendered either as force plot or waterfall plot.
+
+    Parameters
+    ----------
+    train_data: DataFrame
+        training dataset
+    model: TabularPredictor
+        trained AutoGluon predictor
+    rows: pd.DataFrame,
+        rows to explain
+    display_rows: bool, default = False
+        if `True` then display the row before the explanation chart
+    plot: Optional[str], default = 'force'
+        type of plot to visualize the Shapley values. Supported keys:
+        - `force` - Visualize the given SHAP values with an additive force layout
+        - `waterfall` - Visualize the given SHAP values with a waterfall layout
+        - `None` - do not use any visualization
+    baseline_sample: int, default = 100
+        The background dataset size to use for integrating out features. To determine the impact
+        of a feature, that feature is set to "missing" and the change in the model output
+        is observed.
```

Comment:
Very random thought completely tangential to the PR: I wish IDEs/code had logic where we could specify a docstring for a parameter one time in some file, then have it auto-populate via a variable reference in the source-code docs so we just have to write

```
baseline_sample: int, default = 100
     {baseline_sample_docstring}
```

and the IDE magically converts it to text unless we click to force it to show source, so we avoid having to copy/paste the same doc-string many times across many usages.

That is my random thought of the day. Carry on.